PWGGA/GammaConvBase/AliV0ReaderV1.cxx

```diff
@@ -1262,14 +1263,20 @@ Bool_t AliV0ReaderV1::GetAODConversionGammas(){
     if(!fInputGammas){AliError("No Gamma Satellites found");return kFALSE;}
     // Apply Selection Cuts to Gammas and create local working copy
     if(fInputGammas){
+      Bool_t relabelingWorkedForAll = kTRUE;
       for(Int_t i=0;i<fInputGammas->GetEntriesFast();i++){
         gamma=dynamic_cast<AliAODConversionPhoton*>(fInputGammas->At(i));
         if(gamma){
-        if(fRelabelAODs)RelabelAODPhotonCandidates(gamma);
-        if(fConversionCuts->PhotonIsSelected(gamma,fInputEvent)){
-          new((*fConversionGammas)[fConversionGammas->GetEntriesFast()]) AliAODConversionPhoton(*gamma);}
+          if(fRelabelAODs) {
+            relabelingWorkedForAll &= RelabelAODPhotonCandidates(gamma);}
+          if(fConversionCuts->PhotonIsSelected(gamma,fInputEvent)){
+            new((*fConversionGammas)[fConversionGammas->GetEntriesFast()]) AliAODConversionPhoton(*gamma);}
         }
       }
+      if (!relabelingWorkedForAll){
+        AliError("For one ore more photon candidate the AOD daughters could not be found. The labels of those were set to -999999 and the event will get rejected.");
+        return kFALSE;
+      }
```

Comment:
What if  "GetEntries()" in L 1276 returns 0 - meaning we have no V0 candidates? In your fix, this case will also be flagged with event type 2, which must not be the case in my opinion. Could you please check this? We need to be really careful here...

Reply:
Good catch, thanks :) I agree, if we have no V0 candidates (or none of them is passing the cut) we do not necessarily want to flag the event with type 2